lib/posix/strsignal.c

```diff
@@ -0,0 +1,93 @@
+/*
+ * Copyright (c) 2023 Meta
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include "signal.h"
+
+#include <stdio.h>
+
+#include <zephyr/posix/signal.h>
+#include <zephyr/kernel.h>
+
+#ifdef CONFIG_POSIX_SIGNAL_STRING_FULL
+#define SIG_STRING(_signo, _desc) _desc
+#else
+#define SIG_STRING(_signo, _desc) _signo
+#endif
+
+char *strsignal(int signum)
+{
+	static char sigstr[sizeof("RT signal xx")];
+
+	if (!SIGNO_VALID(signum)) {
+		return "Invalid signal";
+	}
+
+	switch (signum) {
```

Comment:
This seems kind of clunky. We should be able to use a LUT. Can you check out what I did for strerror in the minimal libc? If you add the string beside the define as a comment, then you can derive the table at build time.

Reply:
do you mean with a [script](https://github.com/zephyrproject-rtos/zephyr/blob/main/scripts/build/gen_strerror_table.py) to generate that?